src/vendor/cli/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+#
+#  Copyright (c) 2020, The OpenThread Authors.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+set(VENDOR_SOURCES
+    src/vendor_cli_example.c
+)
+
+if(NOT OPENTHREAD_POSIX_VENDOR_CLI_HEADER)
+    set(OPENTHREAD_POSIX_VENDOR_CLI_HEADER "vendor_cli_example.h")
+endif()
+
+add_library(ot-vendor-cli STATIC ${VENDOR_SOURCES})
+
+target_include_directories(ot-vendor-cli
+    PUBLIC
+        include
+    PRIVATE
+        src
+        ${COMMON_INCLUDES}
+)
+
+target_link_libraries(ot-vendor-cli PRIVATE
```

Comment:
I don't understand this list.  What's it doing?

Reply:
`target_link_libraries` defines what libraries should be linked. See the difference between `PRIVATE`, `PUBLIC`, and `INTERFACE` [here](https://cmake.org/cmake/help/latest/command/target_link_libraries.html#libraries-for-a-target-and-or-its-dependents)

So this `target_link_libraries` statement is saying "When compiling `ot-vendor-cli` link the following libs`